boa/src/builtins/number/tests.rs

```diff
@@ -177,12 +178,255 @@ fn to_string() {
     let float_string = forward(&mut engine, "float_string");
     let exp_string = forward(&mut engine, "exp_string");
     let neg_string = forward(&mut engine, "neg_string");
+    let nan_string = forward(&mut engine, "nan_string");
 
     assert_eq!(default_string, String::from("0"));
     assert_eq!(int_string, String::from("123"));
     assert_eq!(float_string, String::from("1.234"));
     assert_eq!(exp_string, String::from("12000"));
     assert_eq!(neg_string, String::from("-1.2"));
+    assert_eq!(nan_string, String::from("NaN"));
```

Comment:
I think we should make this easier to understand like this:
```suggestion
    assert_eq!(&default_string, "0");
    assert_eq!(&int_string, "123");
    assert_eq!(&float_string, "1.234");
    assert_eq!(&exp_string, "12000");
    assert_eq!(&neg_string, "-1.2");
    assert_eq!(&nan_string, "NaN");
```

I think we should do this in all tests.

Reply:
I think for comprehensive tests like we may start to implement something like v8's [mjsunit](https://chromium.googlesource.com/v8/v8/+/3.30.3/test/mjsunit/number-tostring.js?autodive=0). It does not need to be something fancy. Just some file discovery and running them on a basic test suite running them. Also it would be cool to continue running these kind of tests even a previous assert fails. It is kinda annoying to rerun tests on every fix than seeing that a following assert fails.